### PR TITLE
Make references with only editors citable

### PR DESCRIPTION
**Pull request by sdruskat**
**Related issues**

Refs: #334 

**Describe the changes made in this pull request**

- Allow references to have no authors, but editors instead
- Allow reference to have authors and editors
- Test both cases, and additionally add a failing test to assert that the root document still needs authors

**Review checklist**

- [x] Please check if the pull request is against the correct branch  
(format/schema/semantic documentation changes: `develop`; typos, meta files, etc.: `main`)
- [ ] Please check if all changes are recorded in `CHANGELOG.md` and adapt if necessary.
- [x] Please run tests locally.

```bash
cd $(mktemp -d --tmpdir cff.XXXXXX)
git clone https://github.com/citation-file-format/citation-file-format .
git checkout 334-req-authors-or-editors
python3 -m venv env
source env/bin/activate
pip install --upgrade pip wheel setuptools
pip install -r requirements.txt
pytest
<do other things>
```

- [ ] Perhaps test other cases that I might have missed manually

### CODE REVIEW

File: tests/examples-failing/software-editors/CITATION.cff [hunk 1]
```diff
@@ -0,0 +1,12 @@
+editors:
+  - name: The software authors
```
Comment (jspaaks) [inline tests/examples-failing/software-editors/CITATION.cff:2]:
```suggestion
  - name: The software editors
```

**Review by jspaaks — COMMENTED** [review]
I'd like to rename the directories

```shell
git mv tests/examples-failing/software-editors tests/examples-failing/missing-software-authors
```
such that the test name hints at the reason for failure, and 

```shell
git mv tests/examples-passing/editors-no-authors tests/examples-passing/editors-and-authors
```
such that the test name includes the "theme"

File: CHANGELOG.md [hunk 2]
```diff
@@ -35,8 +35,9 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
-- Instances of `person` now explicitly require one of `alias`, `email`, `given-names`, `family-names` or `orcid` to avoid accidentally having empty person objects (notably, `authors`). PR [#462](https://github.com/citation-file-format/citation-file-format/pull/462)
+- Instances of `person` now explicitly require one of `alias`, `email`, `given-names`, `family-names` or `orcid` to avoid accidentally having empty person objects (notably, `authors`); PR [#462](https://github.com/citation-file-format/citation-file-format/pull/462)
 - Fixed the regular expression for ISBN (should only affect ISBN10 strings that didn't have any dashes or spaces, whose check digit is `X`). Issue [#323](https://github.com/citation-file-format/citation-file-format/issues/323); PR [#335](https://github.com/citation-file-format/citation-file-format/pull/335); PR [#337](https://github.com/citation-file-format/citation-file-format/pull/337)
+- Loosen requirement of authors in references to allow for any of authors and/or editors. Issue [#334](https://github.com/citation-file-format/citation-file-format/issues/334); PR [#524](https://github.com/citation-file-format/citation-file-format/pull/524)
```
Comment (jspaaks) [inline CHANGELOG.md:40]: Not sure this qualifies as a fix. From a semver perspective it's more of an addition, right?

**Comment by sdruskat** [comment]
> I'd like to rename the directories

Done in https://github.com/citation-file-format/citation-file-format/pull/524/commits/ade7e95125852767283fd249845b93ab69345546

File: CHANGELOG.md (hunk 2 above)
Comment (sdruskat) [reply CHANGELOG.md:40]: Depending on whether you see the inability to represent edited volumes correctly in `references` as a bug. I think this is a reasonable interpretation, and thus think the change should be in "# Fixes".

Comment (sdruskat) [reply CHANGELOG.md:40]: Decided to move this into **Added** as per semver.

Comment (sdruskat) [reply CHANGELOG.md]: Done in https://github.com/citation-file-format/citation-file-format/pull/524/commits/6de9a66402846bf9459710f5df8edb3f30feaf0f.

**Pull request merged**